### PR TITLE
CI/CD 1005 add traceability to prow

### PR DESCRIPTION
**Pull request by medmes**
**Description**
The traceability of PROW pipelines is very hard because the logs are not providing enough details about actual script/commands/function ran, especially when errors occurs.

**Changes proposed in this pull request**
- Improve the log format to different logs level: info, error, success, banner...
- Improve the lib scripts logging by using the new logging format.
- Adding logging to commands (parent scripts)that uses lib scripts.
- Improve Error handling(logging entire stack trace) by implementing an algorithm that doesn't use trap function.

**Output in case of Info logs:**
Screenshot 2022-05-04 at 22 02 05

**Output in case of error logs:**
Screenshot 2022-05-05 at 19 06 22

**Related issue(s)**
https://github.com/kyma-incubator/reconciler/issues/1005

### CODE REVIEW

**Comment by dekiel** [comment]
@medmes use draft PRs for work in progress.

File: prow/scripts/cluster-integration/reconciler-e2e-nightly-gardener.sh [hunk 1]
```diff
@@ -78,29 +78,30 @@ fi
 # Checks required vars and initializes gcloud/docker if necessary
 gardener::init
 
+log::info "Export cluster name"
```
Comment (jakobmoellersap) [inline prow/scripts/cluster-integration/reconciler-e2e-nightly-gardener.sh:81]: why do we use the log here and not in the functions for reconciler?

File: prow/scripts/lib/log.sh [hunk 2]
```diff
@@ -1,8 +1,22 @@
 #!/usr/bin/env bash
 
-# log::date retruns current date in format expected by logs
+# log::date retrun current date in format expected by logs
```
Comment (jakobmoellersap) [inline prow/scripts/lib/log.sh:3]:
```suggestion
# log::date returns the current date in format expected by logs
```

File: prow/scripts/lib/log.sh [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
 function log::date {
-    date +"%Y/%m/%d %T %Z"
+    date +"%Y-%m-%d %T %Z"
+}
+
+# log::dump_trace prints stacktrace when an error occur
```
Comment (jakobmoellersap) [inline prow/scripts/lib/log.sh:8]:
```suggestion
# log::dump_trace prints stacktrace when an error occurs. 
```

File: prow/scripts/lib/log.sh [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+#
+log::dump_trace() {
+    local frame=1 line func source n=0
+    while caller "$frame"; do
```
Comment (jakobmoellersap) [inline prow/scripts/lib/log.sh:12]: Nice!

File: prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh [hunk 5]
```diff
@@ -46,6 +46,7 @@ function testCustomImage() {
   fi
 }
 
+log::info "Authenticate into GCP"
 gcp::authenticate \
```
Comment (jakobmoellersap) [inline prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh:50]: I think also here we should probably write the logs inside the actual functions

**Review by jakobmoellersap — COMMENTED** [review]
Really cool implementation, I think we should consider putting the logs at the lowest level possible though to make sure that they are always executed and not skipped for pipelines that use the functions but forget to add the log line. Other than that, Really cool! 🚀

**Comment by jakobmoellersap** [comment]
@medmes you still need to sign the cla

File: prow/scripts/provision-vm-compass.sh [hunk 6]
```diff
@@ -38,6 +38,7 @@ function testCustomImage() {
     fi
 }
 
+log::info "Authenticate into GCP"
 gcp::authenticate \
```
Comment (jakobmoellersap) [inline prow/scripts/provision-vm-compass.sh:42]: Can be put into the function

File: prow/scripts/provision-vm-cli.sh [hunk 7]
```diff
@@ -32,6 +32,7 @@ function testCustomImage() {
     fi
 }
 
+log::info "Authenticate into GCP"
```
Comment (jakobmoellersap) [inline prow/scripts/provision-vm-cli.sh:35]: Can be put into the function

**Review by jakobmoellersap — DISMISSED** [review]
To the 🌔

**Pull request merged**